🚨 Deploy Supabase Migrations failed on dev

**Workflow**: Deploy Supabase Migrations
**Branch**: dev
**Commit**: f1fd8d6d4e30cde3e487f4dc0ba78f5d5ad7d79b
**Run**: https://github.com/Mark-Yun/minglit/actions/runs/24082291577
**Triggered by**: feat(tag-discovery): Phase 1 MVP — DB, API, Flutter UI (#1149)

Scheduler: needs-arch-claude-team-1

## Summary

- Tag Discovery Phase 1 MVP 구현: 태그 시스템 도입으로 관심사 기반 이벤트 탐색 지원
- 4 DB 테이블 + RLS + 트리거 + 6 RPC 함수 + 시드 데이터 25개
- partner-manage-party EF에 tag_ids 파라미터 추가 (create/update)
- Tag 모델 + TagRepository + Riverpod providers (minglit_kit)
- 유저 앱: 홈 인기 태그 칩바 + 핫 태그 섹션 + 태그별 이벤트 리스트 + 카드 태그 뱃지
- 파트너 앱: 파티 생성/수정 시 태그 선택 (최대 5개)
- pgTAP 테스트 3개 파일 + EF 테스트 + Flutter 테스트

## Phase 1 범위

| 구현 완료 | Phase 2 (후속) |
|-----------|---------------|
| DB 스키마 4테이블 + RLS + 트리거 | 검색 태그 자동완성 |
| RPC 6개 함수 | 온보딩 관심 태그 선택 |
| EF tag_ids 지원 | 관심사 추천 피드 섹션 |
| 홈 인기 태그 칩바 | OR 필터 (복수 태그 선택 → 피드 필터) |
| 핫 태그 섹션 | |
| 태그별 이벤트 리스트 | |
| 이벤트 카드 태그 뱃지 | |
| 파트너 태그 선택 UI | |

## Test plan

- [ ] pgTAP: 스키마 테스트 (60_tag_discovery_schema_test.sql)
- [ ] pgTAP: RLS 테스트 (61_tag_discovery_rls_test.sql)
- [ ] pgTAP: RPC 테스트 (62_tag_discovery_rpc_test.sql)
- [ ] EF: partner-manage-party tag_ids 테스트
- [ ] Flutter: TagRepository 테스트
- [ ] Flutter: TagEventListController 테스트
- [ ] Flutter: TagSelectionSection 위젯 테스트
- [ ] build_runner 실행 후 analyze/test 통과 확인

Closes #1136

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---------

Co-authored-by: needs-swe-sonnet-subagents-1 <worker@minglit.ai>
Co-authored-by: Claude Opus 4.6 (1M context) <noreply@anthropic.com>
Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
**Actor**: Mark-Yun

**Job Results**:
  ❌ deploy: failure

🤖 **tpm-exec-report-claude-subagents** 분석:

### 근본 원인

`Deploy Supabase Migrations` 워크플로우 **5회 연속 실패** (최근 5 runs 전부 failure).

#1130 수정에서 `dig +short A` → `/etc/hosts` 방식으로 IPv4를 강제했으나, **`dig +short A` 자체가 빈 결과를 반환**하여 실패:
```
##[error]Failed to resolve IPv4 for db.***.supabase.co
```

### 영향

- PR #1149 (Tag Discovery Phase 1) 머지 후 DB 마이그레이션이 dev에 배포되지 않음
- 4개 테이블 + RLS + 트리거 + RPC 함수 미배포 상태

### 수정 방향

`dig +short A` 실패 시 fallback 추가:
```bash
# 방법 1: getent 사용
IPV4=$(getent ahostsv4 "db.${PROJECT_ID}.supabase.co" | head -1 | awk '{print $1}')

# 방법 2: nslookup 사용
IPV4=$(nslookup -type=A "db.${PROJECT_ID}.supabase.co" | grep 'Address:' | tail -1 | awk '{print $2}')

# 방법 3: host 명령
IPV4=$(host -t A "db.${PROJECT_ID}.supabase.co" | awk '/has address/ {print $4; exit}')
```

`needs-swe` 라벨로 라우팅합니다.

--- a/.github/workflows/supabase-deploy.yml
+++ b/.github/workflows/supabase-deploy.yml
@@ -95,10 +95,25 @@ jobs:
           # Fix #1130: gai.conf IPv4 precedence not effective for psql — /etc/hosts override
           # Fix #1153: dig +short A returns empty on some runners — fallback chain
           HOST="db.${PROJECT_ID}.supabase.co"
-          IPV4=$(dig +short A "$HOST" | head -1)
-          : "${IPV4:=$(getent ahostsv4 "$HOST" 2>/dev/null | head -1 | awk '{print $1}')}"
-          : "${IPV4:=$(nslookup -type=A "$HOST" 2>/dev/null | grep 'Address:' | tail -1 | awk '{print $2}')}"
-          : "${IPV4:=$(host -t A "$HOST" 2>/dev/null | awk '/has address/ {print $4; exit}')}"
+          is_valid_ipv4() {
+            [[ $1 =~ ^((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[1-9][0-9]|[0-9])\.){3}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[1-9][0-9]|[0-9])$ ]]
+          }
+          IPV4=""
+          CANDIDATE=$(dig +short A "$HOST" | head -1)
+          is_valid_ipv4 "$CANDIDATE" && IPV4="$CANDIDATE"
+          if [ -z "$IPV4" ]; then
+            CANDIDATE=$(getent ahostsv4 "$HOST" 2>/dev/null | head -1 | awk '{print $1}')
+            is_valid_ipv4 "$CANDIDATE" && IPV4="$CANDIDATE"
+          fi
+          if [ -z "$IPV4" ]; then
+            # Use "^Address: " (space) to exclude the DNS server line "Address:\t8.8.8.8#53" (tab)
+            CANDIDATE=$(nslookup -type=A "$HOST" 2>/dev/null | awk '/^Address: /{print $2}' | tail -1)
+            is_valid_ipv4 "$CANDIDATE" && IPV4="$CANDIDATE"
+          fi
+          if [ -z "$IPV4" ]; then
+            CANDIDATE=$(host -t A "$HOST" 2>/dev/null | awk '/has address/ {print $4; exit}')
+            is_valid_ipv4 "$CANDIDATE" && IPV4="$CANDIDATE"
+          fi
           if [ -z "$IPV4" ]; then
             echo "::error::Failed to resolve IPv4 for ${HOST} (tried dig, getent, nslookup, host)"
             exit 1

--- a/.github/workflows/supabase-deploy.yml
+++ b/.github/workflows/supabase-deploy.yml
@@ -93,13 +93,18 @@ jobs:
           PROJECT_ID: ${{ steps.env.outputs.PROJECT_ID }}
         run: |
           # Fix #1130: gai.conf IPv4 precedence not effective for psql — /etc/hosts override
-          IPV4=$(dig +short A "db.${PROJECT_ID}.supabase.co" | head -1)
+          # Fix #1153: dig +short A returns empty on some runners — fallback chain
+          HOST="db.${PROJECT_ID}.supabase.co"
+          IPV4=$(dig +short A "$HOST" | head -1)
+          : "${IPV4:=$(getent ahostsv4 "$HOST" 2>/dev/null | head -1 | awk '{print $1}')}"
+          : "${IPV4:=$(nslookup -type=A "$HOST" 2>/dev/null | grep 'Address:' | tail -1 | awk '{print $2}')}"
+          : "${IPV4:=$(host -t A "$HOST" 2>/dev/null | awk '/has address/ {print $4; exit}')}"
           if [ -z "$IPV4" ]; then
-            echo "::error::Failed to resolve IPv4 for db.${PROJECT_ID}.supabase.co"
+            echo "::error::Failed to resolve IPv4 for ${HOST} (tried dig, getent, nslookup, host)"
             exit 1
           fi
-          echo "${IPV4} db.${PROJECT_ID}.supabase.co" | sudo tee -a /etc/hosts > /dev/null
-          echo "Resolved db.${PROJECT_ID}.supabase.co → ${IPV4}"
+          echo "${IPV4} ${HOST}" | sudo tee -a /etc/hosts > /dev/null
+          echo "Resolved ${HOST} → ${IPV4}"
 
           REQUIRED=$(jq -r '.vault.required | keys[]' env-manifest.json)
           VAULT_MISSING=""

